sdk/identity/azure-identity/azure/identity/aio/_credentials/authorization_code.py

```diff
@@ -31,13 +32,19 @@ class AuthorizationCodeCredential(object):
     :keyword str client_secret: One of the application's client secrets. Required only for web apps and web APIs.
     """
 
+    async def __aenter__(self):
+        if self._client:
+            await self._client.__aenter__()
+        return self
+
+    async def close(self):
+        """Close the credential's transport session."""
+
+        if self._client:
+            await self._client.__aexit__()
+
     def __init__(
-        self,
-        tenant_id: str,
-        client_id: str,
-        authorization_code: str,
-        redirect_uri: str,
-        **kwargs: "Any"
+        self, tenant_id: str, client_id: str, authorization_code: str, redirect_uri: str, **kwargs: "Any"
     ) -> None:
         self._authorization_code = authorization_code  # type: Optional[str]
```

Comment:
Why optional?

Reply:
Note the private field is optional but the parameter is not. The authorization code is a string, so the private field is initially a string. Authorization codes are single use. After successfully redeeming the code for a token, `get_token` sets the private field to `None` to indicate the credential no longer has an authorization code and can only return a cached token. So, `Optional[str]`.